app/src/main/java/com/example/jwplayerdemo/JWMainActivity.java

```diff
@@ -0,0 +1,148 @@
+import com.example.jwplayerdemo.jwsettings.JWPlayerViewSettings;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+
+public class JWMainActivity extends AppCompatActivity implements
+        JWPlayerViewSettings.OnSettingsListener {
+
+    /*
+     * Instantiate CoordinatorLayout - use the "Hiding Bottom Navigation View Behavior"
+     * */
+    @BindView(R.id.container)
+    CoordinatorLayout container;
+
+    /*
+     * Instantiate FrameLayout a reusable container for Fragments
+     * */
+    @BindView(R.id.framelayout)
+    FrameLayout mFrameLayout;
+
+    /*
+     * Instantiate Bottom Navigation View
+     * */
+    @BindView(R.id.bottom_navigation)
+    BottomNavigationView navigation;
+
+    private Fragment mCurrFragmentView;
+    private Bundle bundle;
+
+    /**
+     * {@link ViewModelProviders} - Preserve Data
+     * The first time the ViewModelProviders.of method is called by JWMainActivity, it creates a new ViewModel instance.
+     * When this method is called again, which happens whenever onCreate is called,
+     * it will return the pre-existing ViewModel associated with the specific Court-Counter JWMainActivity.
+     * This is what preserves the data.
+     * <p>
+     * {@link - https://developer.android.com/training/basics/fragments/communicating}
+     * {@link - https://medium.com/androiddevelopers/viewmodels-a-simple-example-ed5ac416317e}
+     *
+     * @see androidx.lifecycle.ViewModel
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // Create a ViewModel the first time the system calls an activity's onCreate() method.
+        // Re-created activities receive the same MyViewModel instance created by the first activity.
+        ViewModelProviders
+                .of(this)
+                .get(JWViewModel.class);
+
+        // Bind all of my views to this activity
+        // TODO: I might get rid of bind because I'm not sure if Butterknife will stay up to date
+        ButterKnife.bind(this);
+
+        // Call Support Fragment's Manager to manage my fragments
+        FragmentManager fragmentManager = getSupportFragmentManager();
+
+        // Set OnNavigationItemListener
+        navigation.setOnNavigationItemSelectedListener(menuItem -> {
+
+            switch (menuItem.getItemId()) {
+                case R.id.navigation_jwplayer:
+                    Log.i("HYUNJOO", "JWMainActivity - navigation bar - jwplayerview");
+                    mCurrFragmentView = new JWPlayerViewExample();
+                    mCurrFragmentView.setArguments(bundle);
+                    break;
+                case R.id.navigation_settings:
+                    Log.i("HYUNJOO", "JWMainActivity - navigation bar - settings");
+                    mCurrFragmentView = new JWPlayerViewSettings();
```

Comment:
does there need to be a `setArguments()` call her for the `JWPlayerViewSettings` instance?  or is it getting its state some other way?

Reply:
I think I should remove the bundle because JWPlayerView and JWPlayerViewSettings should always receive the updated information from the store, right?